planner/core/planbuilder.go

```diff
@@ -1619,44 +1620,108 @@ func (b *PlanBuilder) buildLoadStats(ld *ast.LoadStatsStmt) Plan {
 	return p
 }
 
-func (b *PlanBuilder) buildSplitIndexRegion(node *ast.SplitIndexRegionStmt) (Plan, error) {
+const maxSplitRegionNum = 1000
+
+func (b *PlanBuilder) buildSplitRegion(node *ast.SplitRegionStmt) (Plan, error) {
 	tblInfo := node.Table.TableInfo
-	indexInfo := tblInfo.FindIndexByName(strings.ToLower(node.IndexName))
+	indexInfo := tblInfo.FindIndexByName(node.IndexName.L)
 	if indexInfo == nil {
 		return nil, ErrKeyDoesNotExist.GenWithStackByArgs(node.IndexName, tblInfo.Name)
 	}
+	mockTablePlan := LogicalTableDual{}.Init(b.ctx)
+	schema := expression.TableInfo2SchemaWithDBName(b.ctx, node.Table.Schema, tblInfo)
+	mockTablePlan.SetSchema(schema)
 
-	indexValues := make([][]types.Datum, 0, len(node.ValueLists))
-	for i, valuesItem := range node.ValueLists {
-		if len(valuesItem) > len(indexInfo.Columns) {
-			return nil, ErrWrongValueCountOnRow.GenWithStackByArgs(i + 1)
-		}
-		valueList := make([]types.Datum, 0, len(valuesItem))
+	convertValue2ColumnType := func(valuesItem []ast.ExprNode) ([]types.Datum, error) {
```

Comment:
can you split this function outside the definition of `buildSplitRegion()`? It's a complicated function with 40 lines, which reduces the readability of `buildSplitRegion()`